ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -139,12 +139,25 @@ def wrapped(*args, **kwargs):
 
 def prepare_reverse_zone(host, ip):
     zone = get_reverse_zone_default(ip)
-    result = host.run_command(["ipa",
-                      "dnszone-add",
-                      zone], raiseonerr=False)
-    if result.returncode > 0:
-        logger.warning("%s", result.stderr_text)
-    return zone, result.returncode
+    validate = host.run_command(
+        [
+            "ipa",
+            "dnszone-find",
+            zone
+        ],
+        raiseonerr=False)
+    if validate.returncode > 0:
+        result = host.run_command(
+            [
+                "ipa",
+                "dnszone-add",
+                "--skip-overlap-check",  # to force dnszone-add
+                zone
+            ],
+            raiseonerr=False)
+        if result.returncode > 0:
+            logger.warning("%s", result.stderr_text)
+        return zone, result.returncode
```

Comment:
If the ipa dnszone-find command succeeds (validate.returncode == 0), the function prepare_reverse_zone exits without returning any value, while it should return a Tuple (string, int). This will cause unpacking exception if the caller tries to read the returned value.